### DISCUSSION TITLE
Scale down objects on zoom in

### DISCUSSION THREAD

**Discussion by JiahuiChen99** [fabricjs/fabric.js#8516] [Q&A]
Hi there again 👋 ,

I'd like to scale down the objects of the canvas on zoom in. Right now I'm scaling the objects like this:

```js
object.scale(
  1 / canvas.getZoom()
)

// or

object.set({
   scaleX: 1 / canvas.getZoom(),
   scaleY: 1 / canvas.getZoom()
})
```

The objects get scaled, however if I move them and perform a zoom action, these **disappear** 🤯 . 
If I retrieve the objects of the canvas (canvas.getObjects()), they're still "there". As if these went invisible.

**Comment by iocoker** [comment]
@JiahuiChen99 It's possible the scaling is applied to objects from `left/top` origins. So you may need to offset the positions `(left/top)` by +- a factor of the scale value. Try checking their left/top positions after the zoom to give you an idea of where they are on the canvas.
It's also possible the canvas was zoomed as well and they got clipped out of view.

It will be nice if you could post a sample [JSFiddle](https://jsfiddle.net/) to give an idea of what's happening.

Btw, why don't you want the entire canvas scaled/zoomed instead?

**Reply by JiahuiChen99** [reply]
Hi @iocoker
After some time of debugging, I realized that the objects are positioned correctly, and they don't **disappear**. 
Seems like moving the object around, messes up with the scale, width and height of the object, these end up having sub-pixel values.

This is the current behaviour: 
Zoom In / Out: Sets the object scale to something < 1
Move the object around: Resets the scaleX, scaleY to 1

Example: 
Start with an object with width & height of 7
![image](https://user-images.githubusercontent.com/37770349/208692923-9912247b-8114-41bb-9e67-da895680fcda.png)
Zoom In to value 5. The object width & height are still 7. The scales values are set to 0.2 because of 1 / 5.
![image](https://user-images.githubusercontent.com/37770349/208693182-9eeb8ad9-a6da-4685-a36e-39ce7281ecb2.png)
Moved the objects a little bit. Top and left values have changed a little bit. The scales values reseted back to 1. But the width and height of the object got shrinked 🤯 
![image](https://user-images.githubusercontent.com/37770349/208693999-77d2ccbe-354a-420e-b7db-6a58df5a7519.png)

**Reply by JiahuiChen99** [reply]
Seems like this behaviour was due to a hacky fix I made for this issue #8492. While interacting with the group of objects, it recalculates the bounding box (to avoid a weird glitch on masive zoom) and the method resets the scales back to 1. That's why this problem was happening.
